nautobot/extras/tests/test_api.py

```diff
@@ -3475,6 +3478,149 @@ def setUpTestData(cls):
+        self.assertEqual(member_count, len(response.json()["results"]))
+        # TODO: assert that members are serialized correctly?
+
+
+class StaticGroupAssociationTest(APIViewTestCases.APIViewTestCase):
+    model = StaticGroupAssociation
+    choices_fields = ["associated_object_type"]
+
+    # StaticGroupAssociation records created for SoftwareImageFile records will contain a `hashing_algorithm` key;
+    # presence of strings like "md5" and "sha256" in the API response for StaticGroupAssociation is *not* a failure
+    VERBOTEN_STRINGS = ("password",)
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.sg1 = StaticGroup.objects.create(name="Locations", content_type=ContentType.objects.get_for_model(Location))
+        cls.sg2 = StaticGroup.objects.create(name="Devices", content_type=ContentType.objects.get_for_model(Device))
+        cls.sg3 = StaticGroup.objects.create(name="VLANs", content_type=ContentType.objects.get_for_model(VLAN))
+        location_pks = list(Location.objects.values_list("pk", flat=True)[:4])
+        device_pks = list(Device.objects.values_list("pk", flat=True)[:4])
+        StaticGroupAssociation.objects.create(
+            static_group=cls.sg1,
+            associated_object_type=ContentType.objects.get_for_model(Location),
+            associated_object_id=location_pks[0],
+        )
+        StaticGroupAssociation.objects.create(
+            static_group=cls.sg1,
+            associated_object_type=ContentType.objects.get_for_model(Location),
+            associated_object_id=location_pks[1],
+        )
+        StaticGroupAssociation.objects.create(
+            static_group=cls.sg2,
+            associated_object_type=ContentType.objects.get_for_model(Device),
+            associated_object_id=device_pks[0],
+        )
+        StaticGroupAssociation.objects.create(
+            static_group=cls.sg2,
+            associated_object_type=ContentType.objects.get_for_model(Device),
+            associated_object_id=device_pks[1],
+        )
+
+        cls.create_data = [
+            {
+                "static_group": cls.sg1.pk,
+                "associated_object_type": "dcim.location",
+                "associated_object_id": location_pks[2],
+            },
+            {
+                "static_group": cls.sg1.pk,
+                "associated_object_type": "dcim.location",
+                "associated_object_id": location_pks[3],
+            },
+            {
+                "static_group": cls.sg2.pk,
+                "associated_object_type": "dcim.device",
+                "associated_object_id": device_pks[2],
+            },
+            {
+                "static_group": cls.sg2.pk,
+                "associated_object_type": "dcim.device",
+                "associated_object_id": device_pks[3],
+            },
+        ]
+        # TODO: this isn't really valid since we're changing the associated_object_type but not the associated_object_id
```

Comment:
How do we deal with invalid data passed to bulk update on other classes?

Reply:
If it's truly invalid (would get rejected by bulk-update API) that's one thing - in this case though we're just making the GFK point to a nonexistent object, which is possible in the DB unless we'd have to have a custom clean() or validate() method to catch it. Might be worth adding?

Reply:
Probably not worth the effort for this issue imo